### PR TITLE
Move I/O Frontend Code

### PR DESCRIPTION
**Pull request by tabathadelane**
The site builds locally with a few bugs:  

- ~~when clicking on a category, the page breaks. However, the filtered page displays when you refresh.~~ Resolved
- ~~Codestream appears in the wrong place?~~ Resolved 

There is still some work that can be done to pull out unnecessary code. 

Relates to https://github.com/newrelic/developer-website/issues/2003

### CODE REVIEW

**Comment by aswanson-nr** [comment]
Still need to figure out related resources

**Comment by aswanson-nr** [comment]
This should be merged after #7

**Comment by aswanson-nr** [comment]
Currently has an issue, the 404 page from the theme uses the GlobalHeader. So we will need to create a new 404 page for this site.

File: src/components/GuidedInstallTile.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import React from 'react';
```
Comment (aswanson-nr) [inline src/components/GuidedInstallTile.js:1]: I don't think we're using this after the redesign, can probably be removed.

File: gatsby-config.js [hunk 2]
```diff
@@ -20,6 +155,34 @@ module.exports = {
         typeName: "Quickstarts",
       },
     },
+    // "gatsby-remark-images",
```
Comment (aswanson-nr) [inline gatsby-config.js:158]: Delete these commented out plugins

File: src/components/GuidedInstallTile.js (hunk 1 above)
Comment (aswanson-nr) [reply src/components/GuidedInstallTile.js:1]: Going to leave this for now to reduce the number of changes

File: gatsby-config.js [hunk 3]
```diff
@@ -1,31 +1,168 @@
...
+          languages: ['yaml', 'sass', 'scss', 'java'],
+        },
+        relatedResources: {
+          swiftype: {
+            resultsPath: `${__dirname}/src/data/related-pages.json`,
+            refetch: Boolean(process.env.BUILD_RELATED_CONTENT),
+            engineKey: 'Ad9HfGjDw4GRkcmJjUut',
+            limit: 5,
+            getSlug: ({ node }) => {
+              if (node.internal.type === 'Mdx') {
```
Comment (LizBaker) [inline gatsby-config.js:43]: are there ever going to be mdx files in the IO site?

File: gatsby-config.js [hunk 4]
```diff
@@ -1,31 +1,168 @@
...
+        },
+        newrelic: {
+          configs: {
+            production: {
+              instrumentationType: 'proAndSPA',
+              accountId: '10956800',
+              trustKey: '1',
+              agentID: '30712246',
+              licenseKey: 'NRJS-649173eb1a7b28cd6ab',
+              applicationID: '30712246',
```
[the remark this answers is not in the document]
Comment (aswanson-nr) [reply gatsby-config.js:95]: Yes! Got all that happening in https://github.com/newrelic/developer-website/issues/2006

File: gatsby-config.js [hunk 5]
```diff
@@ -1,31 +1,168 @@
...
+              beacon: 'staging-bam-cell.nr-data.net',
+              errorBeacon: 'staging-bam-cell.nr-data.net',
+            },
+          },
+        },
+        tessen: {
+          tessenVersion: '1.14.0',
+          product: 'DEV',
+          subproduct: 'TDEV',
```
[the remark this answers is not in the document]
Comment (aswanson-nr) [reply gatsby-config.js:114]: yes! That work is happening over here! https://github.com/newrelic/developer-website/issues/2005

File: gatsby-config.js [hunk 6]
```diff
@@ -1,31 +1,168 @@
...
+          segmentWriteKey: 'Ako0hclX8WGHwl9rm4n5uxLtT4wgEtuU',
```
[the remark this answers is not in the document]
Comment (aswanson-nr) [reply gatsby-config.js:115]: We're actually going to continue to report into the dev segment source, but we will be updated the product and subproduct. That's all in this ticket https://github.com/newrelic/developer-website/issues/2005

File: gatsby-config.js [hunk 7]
```diff
@@ -1,31 +1,168 @@
...
+              } else if (node.internal.type === 'Quickstarts') {
+                return resolveQuickstartSlug(node.name, node.id);
+              }
+            },
+            getParams: ({ node }) => {
+              let tags = [];
+              let title = '';
+              if (node.frontmatter) {
+                tags = node.frontmatter.tags;
+                title = node.frontmatter.title;
```
[the remark this answers is not in the document]
Comment (aswanson-nr) [reply gatsby-config.js]: Good call, removed!

File: gatsby-config.js [hunk 8]
```diff
@@ -1,31 +1,168 @@
...
+                      '!views_page_menu',
+                      '!term_page_api_menu',
+                      '!term_page_landing_page',
+                    ],
+                  },
+                },
+              };
+            },
+            filter: ({ node }) =>
+              node.frontmatter?.template === 'GuideTemplate' ||
```
[the remark this answers is not in the document]
Comment (aswanson-nr) [reply gatsby-config.js]: Removed!

File: gatsby-config.js (hunk 3 above)
Comment (aswanson-nr) [reply gatsby-config.js]: Nope! Removed

**Review by josephgregoryii — APPROVED** [review]

**Pull request merged**